daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -138,6 +138,7 @@ class Engine(val config: EngineConfig = EngineConfig.Stable) {
       nodeSeed: Option[crypto.Hash],
       submissionTime: Time.Timestamp,
       ledgerEffectiveTime: Time.Timestamp,
+      checkAuthorization: Boolean = true,
```

Comment:
```suggestion
      checkAuthorization: CheckAuthorizationMode = CheckAuthrorizationMode.On,
```

with somewhere in this file

```scala
sealed abstract class CheckAuthorizationMode extends Product with Serializable
object CheckAuthorizationMode{
  case object On extends CheckAuthorizationMode
  @deprecated("Backward compatibility flag will be dropped soon", "1.6.0")
  case object Off extends CheckAuthorizationMode
}
```

Reply:
Avoiding boolean blindness! Thanks.

Reply:
Done. except for the `@deprecated`. DIscussed with @remy, and he is happy with that.